components/core/src/ffi/ir_stream/decoding_methods.cpp

```diff
@@ -409,6 +423,71 @@ namespace ffi::ir_stream {
         return IRErrorCode_Success;
     }
 
+    template <typename encoded_variable_t>
+    static std::string decode_message (
+            string_view logtype,
+            vector <encoded_variable_t> encoded_vars,
+            vector <string_view> dictionary_vars
+    ) {
+        std::string message;
+        size_t encoded_vars_length = encoded_vars.size();
+        size_t dict_vars_length = dictionary_vars.size();
+        size_t next_static_text_begin_pos = 0;
+
+        size_t dictionary_vars_ix = 0;
+        size_t encoded_vars_ix = 0;
+        for (size_t cur_pos = 0; cur_pos < logtype.length(); ++cur_pos) {
+            auto c = logtype[cur_pos];
+            if (enum_to_underlying_type(VariablePlaceholder::Float) == c) {
```

Comment:
Since all the comparisons are against `c`, what do you think about using a switch-case instead?